compiler/rustc_target/src/spec/mod.rs

```diff
@@ -2076,6 +2076,42 @@ enum TargetKind {
     Builtin,
 }
 
+#[derive(PartialEq, Clone, Debug)]
+pub enum BinaryFormat {
+    Coff,
+    Elf,
+    MachO,
+    Wasm,
+    Xcoff,
+}
+
+impl FromStr for BinaryFormat {
+    type Err = ();
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Coff" => Ok(Self::Coff),
+            "Elf" => Ok(Self::Elf),
+            "MachO" => Ok(Self::MachO),
+            "Wasm" => Ok(Self::Wasm),
+            "Xcoff" => Ok(Self::Xcoff),
```

Comment:
```suggestion
            "coff" => Ok(Self::Coff),
            "elf" => Ok(Self::Elf),
            "mach-o" => Ok(Self::MachO),
            "wasm" => Ok(Self::Wasm),
            "xcoff" => Ok(Self::Xcoff),
```
I'd name the strings like this in lowercase for consistency with the other values. Additionally, please move this to https://github.com/rust-lang/rust/blob/37b2ebc676e0024dbb376ed7a2ba4c50c94c1661/compiler/rustc_target/src/spec/mod.rs#L1645 right next to all the other structs.